### DISCUSSION TITLE
Invalid Search Limit parameter in java API

### DISCUSSION THREAD

**Discussion by luliloveshe** [milvus-io/milvus#26671] [Q&A and General discussion]
Invalid Search Limit parameter in java API
![DF76YS5B@%11JQX@E1LXV8A](https://github.com/milvus-io/milvus/assets/32316248/e382b23e-728e-4df5-a5e7-9e0c022a989c)
![V1N`FKB${BQ@(BPW2H5ELMT](https://github.com/milvus-io/milvus/assets/32316248/cbddf097-790b-445a-9424-91e7d1b80f84)

**Comment by yhmo** [comment]
Good catch. 
"The limit for pagination with limit as the key and an integer as the value." ------- this sentence is an out-of-date description, the SearchParam.withTopK() has defined the "limit" value, no need to provide it in withParams()
We will update the doc soon.

**Reply by luliloveshe** [reply]
In this case, milvus does not provide paging functionality. Pagination can only be done on the Java side based on the number of topK returns.

**Comment by luliloveshe** [comment]
If limit is invalid, what is offset used for?

**Reply by yhmo** [reply]
In fact, limit/offset is not pagination.
Assume you search with topk=100, offset =30, the results will contain 70 items.
Assume you search with topk=10, offset =30, the results will contain 0 items.
The meaning of limit=100/offset=30 is: find out 100 nearest items, and return items. from No.30 to No.100

If you didn't set the offset, it will return 100 items. If you set offset=50, it returns 50 items; set offset=90, it returns 10 items, etc.
The "limit" is set by SearchParam.withTopk().

**Reply by yhmo** [reply]
A pagination-like feature is "range search" ------- https://milvus.io/docs/within_range.md#Configure-a-range-for-vector-filtering

**Comment by luliloveshe** [comment]
Thank you so much
